readme.txt
=== Lightweight Grid Columns ===
Contributors: edge22
Donate link: https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=UVPTY2ZJA68S6
Tags: columns, columns shortcode, grid columns
Requires at least: 4.0
Tested up to: 4.2.2
Stable tag: 0.4
License: GPLv2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.html

Easily add desktop, tablet and mobile friendly columns to your content using an easy to use shortcode.

== Description ==

Lightweight Grid Columns are super easy to use! Install and activate the plugin, then look for the columns icon within your TinyMCE toolbar (see screenshots).

Specify your desired desktop, tablet and mobile widths of the columns, add your content and then insert your shortcode into your content.

Lightweight Grid Columns uses the awesome Unsemantic Framework: (http://unsemantic.com/)

Check out GeneratePress, our awesome WordPress theme! (http://wordpress.org/themes/generatepress)

= Features include: =

* Desktop grid width
* Tablet grid width
* Mobile grid width
* Add custom classes
* Add custom inline styles

== Installation ==

There's two ways to install Lightweight Grid Columns.

1. Go to "Plugins > Add New" in your Dashboard and search for: Lightweight Grid Columns
2. Download the .zip from WordPress.org, and upload the folder to the `/wp-content/plugins/` directory via FTP.

In most cases, #1 will work fine and is way easier.

== Frequently Asked Questions ==

= How do I add the shortcodes? =

* Make sure Lightweight Grid Columns is activated.
* While editing your post or page, go into the "Visual Editor".
* Look for the columns icon (see screenshots tab).
* Choose your desktop, tablet and mobile grid widths.
* If you're adding the last column in a row, check the "Last" checkbox (this is important).

= What if I don't want to use the TinyMCE button? =

Fair enough! Simply add the shortcode to your content.

For example, the below will output 4 columns on desktop, 2 columns on tablet and 1 column on mobile

	[lgc_column grid="25" tablet_grid="50" mobile_grid="100"]Some content[/lgc_column]
	
	[lgc_column grid="25" tablet_grid="50" mobile_grid="100"]Some content[/lgc_column]
	
	[lgc_column grid="25" tablet_grid="50" mobile_grid="100"]Some content[/lgc_column]
	
	[lgc_column grid="25" tablet_grid="50" mobile_grid="100" last="true"]Some content[/lgc_column]
	
Take note of our last column - see the last="true" part? Don't forget ;)

= What are my options for percentages? =

This is taken directly from (http://unsemantic.com):

There are grid classes named grid-x where "x" is a number that represents the percentage width of each grid unit. These cover multiples of 5, up to 100 (grid-5, grid-10 ... grid-95, grid-100). There are also classes for dividing a page into thirds: grid-33 and grid-66 which are 33.3333% and 66.6667% wide, respectively. 

= Are there any other options I can use in the shortcode? =

Yes!

You can use:

= class =
	
	[lgc_column grid="25" tablet_grid="50" mobile_grid="100" class="push-25"]Some content[/lgc_column]

= style =

	[lgc_column grid="25" tablet_grid="50" mobile_grid="100" style="padding-left:0px;"]Some content[/lgc_column]

== Screenshots ==

1. The columns icon in your TinyMCE editor.
2. The options inside the columns shortcode generator.
3. A look at some awesome columns!

== Changelog ==

= 0.4 =
* Show TinyMCE icon on all post types

= 0.3 =
* Fix bug where TinyMCE icon wasn't showing up if SiteOrigin Page Builder and Black Studio TinyMCE Widget were activated together
* Add IE 7 & 8 compatibility

= 0.2 =
* Allow line breaks inside columns

= 0.1 =
* Initial release

== Upgrade Notice ==

= 0.4 =
Show TinyMCE icon on all post types

= 0.3 =
Add old IE compatibility

= 0.2 =
Allow line breaks inside your columns

= 0.1 =
* Initial release

changelog.txt
Changelog Legend:
----

[+] = Added
[*] = Changed
[^] = Moved
[=] = No Changes
[x] = Deleted
[!] = Bugs

----

(6/20/2015) - 0.4
[*] Show TinyMCE icon on all post types

----

(6/19/2015) - 0.3
[!] Fix bug where TinyMCE icon wasn't showing up if SiteOrigin Page Builder and Black Studio TinyMCE Widget were activated together
[+] Add IE 7 & 8 compatibility

----

(5/21/2015) - 0.2
[*] Allow line breaks inside columns

----

(5/15/2015) - 0.1
Initial Release

lightweight-grid-columns.php
<?php
/*
Plugin Name: Lightweight Grid Columns
Plugin URI: http://generatepress.com
Description: Add lightweight grid columns to your content using easy to use shortcodes.
Version: 0.4
Author: Thomas Usborne
Author URI: http://edge22.com
License: GNU General Public License v2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.html
*/

/**
 * Load plugin textdomain.
 *
 * @since 0.1
 */
add_action( 'plugins_loaded', 'lgc_load_textdomain' );
function lgc_load_textdomain() {
  load_plugin_textdomain( 'lgc', false, dirname( plugin_basename( __FILE__ ) ) . '/languages/' ); 
}

if ( ! function_exists( 'lgc_shortcodes_register_shortcode' ) ) :
/*
 * Declare our shortcode
 */
add_action( 'init','lgc_shortcodes_register_shortcode' );
function lgc_shortcodes_register_shortcode()
{
	add_shortcode( 'lgc_column', 'lgc_columns_shortcode' );
}
endif;

if ( ! function_exists( 'lgc_add_shortcode_button' ) ) :
/*
 * Set it up so we can register our TinyMCE button
 */
add_action('admin_init', 'lgc_add_shortcode_button');
function lgc_add_shortcode_button() 
{
	// check user permissions
	if ( !current_user_can( 'edit_posts' ) && !current_user_can( 'edit_pages' ) )
		return;
		
	// check if WYSIWYG is enabled
	if ( get_user_option( 'rich_editing' ) == 'true') {
		add_filter( 'mce_external_plugins', 'lgc_shortcodes_add_tinymce_plugin' );
		add_filter( 'mce_buttons', 'lgc_shortcodes_register_button' );
	}
}
endif;

if ( ! function_exists( 'lgc_shortcodes_add_tinymce_plugin' ) ) :
/*
 * Register our tinyMCE button javascript
 */
function lgc_shortcodes_add_tinymce_plugin( $plugin_array ) {
	$plugin_array[ 'lgc_shortcodes_button' ] = plugins_url( '/js/button.js', __FILE__ );
	return $plugin_array;
}
endif;

if ( ! function_exists( 'lgc_shortcodes_register_button' ) ) :
/*
 * Register our TinyMCE button
 */
function lgc_shortcodes_register_button( $buttons ) {
	array_push( $buttons, 'lgc_shortcodes_button' );
	return $buttons;
}
endif;

if ( ! function_exists( 'lgc_translatable_strings' ) ) :
add_action( 'admin_head','lgc_translatable_strings', 0 );
function lgc_translatable_strings()
{
	?>
	<script type="text/javascript">
		var lgc_add_columns = '<?php _e( 'Add columns', 'lgc' ); ?>';
		var lgc_columns = '<?php _e( 'Columns', 'lgc' ); ?>';
		var lgc_desktop = '<?php _e( 'Desktop grid percentage', 'lgc' ); ?>';
		var lgc_tablet = '<?php _e( 'Tablet grid percentage', 'lgc' ); ?>';
		var lgc_mobile = '<?php _e( 'Mobile grid percentage', 'lgc' ); ?>';
		var lgc_content = '<?php _e( 'Content', 'lgc' ); ?>';
		var lgc_last = '<?php _e( 'Last column in row?', 'lgc' ); ?>';
	</script>
	<?php
}
endif;

if ( ! function_exists( 'lgc_shortcodes_admin_css' ) ) :
/*
 * Add our admin CSS
 */
add_action( 'admin_enqueue_scripts', 'lgc_shortcodes_admin_css' );
function lgc_shortcodes_admin_css() {
	wp_enqueue_style( 'lgc-columns-admin', plugins_url('/css/admin.css', __FILE__) );
}
endif;

if ( ! function_exists( 'lgc_shortcodes_css' ) ) :
/*
 * Add the unsemantic framework
 */
add_action( 'wp_enqueue_scripts', 'lgc_shortcodes_css', 99 );
function lgc_shortcodes_css() {
	wp_enqueue_style( 'lgc-unsemantic-grid-responsive-tablet', plugins_url('/css/unsemantic-grid-responsive-tablet.min.css', __FILE__) );
}
endif;

if ( ! function_exists( 'lgc_columns_shortcode' ) ) :
/*
 * Create the output of the columns shortcode
 */
function lgc_columns_shortcode( $atts , $content = null ) {

	extract( shortcode_atts(
		array(
			'grid' => '50',
			'tablet_grid' => '50',
			'mobile_grid' => '100',
			'last' => '',
			'class' => '',
			'style' => '',
		), $atts )
	);
	
	$clear = ( 'true' == $last ) ? '<div class="lgc-clear"></div>' : '';
	$inlineCSS = ( ! empty( $style ) ) ? ' style="' . esc_attr( $style ) . '"' : '';
	return lgc_columns_helper( '<div class="lgc-column lgc-grid-parent lgc-grid-' . intval( $grid ) . ' lgc-tablet-grid-' . intval( $tablet_grid ) . ' lgc-mobile-grid-' . intval( $mobile_grid ) . '"><div class="inside-grid-column ' . esc_attr( $class ) . '"' . $inlineCSS . '>' . $content . '</div></div>' . $clear );

}
endif;

if ( ! function_exists( 'lgc_columns_helper' ) ) :
function lgc_columns_helper( $content ){   
    
	$array = array (
        '<p>[' => '[', 
        ']</p>' => ']', 
        ']<br />' => ']'
    );

    $content = strtr( $content, $array );
    return do_shortcode( shortcode_unautop( force_balance_tags( trim( $content ) ) ) );
	
}
endif;

if ( ! function_exists( 'lgc_ie_compatibility' ) ) :
/** 
 * Add compatibility for IE8 and lower
 * @since 0.3
 */
add_action('wp_head','lgc_ie_compatibility');
function lgc_ie_compatibility()
{
?>
	<!--[if lt IE 9]>
		<link rel="stylesheet" href="<?php echo plugins_url('/css/ie.min.css', __FILE__); ?>" />
	<![endif]-->
<?php
}
endif;